src/test/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecutionTest.java

```diff
@@ -348,6 +354,25 @@ private static void assertStepExecutions(FlowExecution e, String... steps) throw
         });
     }
 
+    @Issue("JENKINS-46961")
+    @Test public void interruptedWhileStarting() throws Exception {
+        story.then(r -> {
+            WorkflowJob p = r.createProject(WorkflowJob.class, "p");
+            p.setDefinition(new CpsFlowDefinition(
+                "import groovy.transform.*\n" +
+                "import hudson.model.Executor\n" +
+                "import hudson.model.Result\n" +
+                // This runs during CpsFlowExecution.parseScript which runs during CpsFlowExecution.start.
+                "@ASTTest(value={ Executor.currentExecutor().interrupt(Result.ABORTED) }) _\n", false));
+            r.buildAndAssertStatus(Result.FAILURE, p);
+        });
+        story.then(r -> {
+            WorkflowJob p = r.jenkins.getItemByFullName("p", WorkflowJob.class);
+            WorkflowRun b = p.getLastBuild();
+            assertFalse("Build should be completed", b.isBuilding());
```

Comment:
Currently fails here, because `b.isBuilding()` is true after the restart (it is false before the restart).